src/context/ExtraInfoState.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import * as React from 'react';
+
+const initialState = {
+  isBaggageVisible: false,
+};
+
+type Props = {
+  children: React.Node,
+};
+
+export type State = {
+  isBaggageVisible: boolean,
+};
+
+export type ExtraInfoStateType = State & {
+  toggleBaggage: () => void,
+};
+
+export const ExtraInfoState = React.createContext({
+  ...initialState,
+  toggleBaggage: () => {}, // eslint-disable-line no-unused-vars
+});
+
+class ExtraInfoStateProvider extends React.Component<Props, State> {
+  state = initialState;
+
+  toggleBaggage = () => {
+    this.setState(prevState => ({
+      isBaggageVisible: !prevState.isBaggageVisible,
+    }));
+  };
+
+  render() {
+    const { isBaggageVisible } = this.state;
+
+    return (
+      <ExtraInfoState.Provider
+        value={{ isBaggageVisible, toggleBaggage: this.toggleBaggage }}
```

Comment:
FYI: there is one caveat when using React context, see this: https://reactjs.org/docs/context.html#caveats I'm already fixing it in #494